src/name.rs

```diff
@@ -77,20 +79,33 @@ impl<'a> From<DNSNameRef<'a>> for DNSName {
 #[derive(Clone, Copy)]
 pub struct DNSNameRef<'a>(untrusted::Input<'a>);
 
+/// An error indicating that a `DNSNameRef` could not built from the provided input.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct InvalidDNSNameError;
+
+impl fmt::Display for InvalidDNSNameError {
```

Comment:
This needs to be under a `#[cfg(feature = "std")]`. This is why I prefer we fully qualify things like `impl std::fmt::Display`.